src/nxdoc/nuxeo-server/administration/observability/metrics.md

```diff
@@ -0,0 +1,527 @@
+- `nuxeo.importer.stream.consumer.batchCommit`: Timer on batch transaction commit
+- `nuxeo.importer.stream.consumer.batchFailure`: Counter of batch failure
+
+### Nuxeo Instance UP
+
+Each time a Nuxeo node is started it reports the following metrics that can be
+aggregated to count the number of Nuxeo nodes in the cluster.
+
+- `nuxeo.instance-up`: One if up
+
+### Nuxeo Repositories
+
+Repository metrics have a `repository` dimension with the name of the repository.
+
+- `nuxeo.repositories.repository.documents.create`: Count the number of documents created since the node is up
+- `nuxeo.repositories.repository.documents.delete`: Count the number of documents deleted since the node is up
+- `nuxeo.repositories.repository.documents.update`: Count the number of documents updated since the node is up
+
+- `nuxeo.repositories.repository.query`: Timer on NXQL queries
+- `nuxeo.repositories.repository.save`: Timer on session saves
+- `nuxeo.repositories.sessions`: Counter of active sessions
+
+- `nuxeo.repositories.repository.connection.count`: Number of connections used
+- `nuxeo.repositories.repository.connection.idle`: Number of connections idle
+- `nuxeo.repositories.repository.connection.killed`: Killed connections
+
+Repository document cache stats:
+- `nuxeo.repositories.repository.cache.eviction.count`
+- `nuxeo.repositories.repository.cache.hit.count`
+- `nuxeo.repositories.repository.cache.hit.ratio`
+- `nuxeo.repositories.repository.cache.miss.count`
+- `nuxeo.repositories.repository.cache.read.count`
+- `nuxeo.repositories.repository.cache.size`
+
+Repository children documents cache stats:
+- `nuxeo.repositories.repository.childCache.eviction.count`
+- `nuxeo.repositories.repository.childCache.hit.count`
+- `nuxeo.repositories.repository.childCache.hit.ratio`
+- `nuxeo.repositories.repository.childCache.miss.count`
+- `nuxeo.repositories.repository.childCache.read.count`
+- `nuxeo.repositories.repository.childCache.size`
+
+### Nuxeo Streams
+
+[Nuxeo Stream]({{page page='nuxeo-stream'}}) Processor relies on a topology of computations.
+It is important to monitor failure [because a manual intervention is required]({{page page='nuxeo-stream'}}#recovery).
+
+- `nuxeo.streams.failure`: Total number of computation threads terminated because of failure for the node (all computations)
+
+Each computation has the following metrics, the computation name being a dimension, for instance `audit/writer`, `bulk/csvExport` ...
+
+- `nuxeo.streams.computation.failure`: Count the number of computation threads terminated by a failure
+- `nuxeo.streams.computation.skippedRecord`: Count the number of record skipped after failure, the computation thread is not terminated
+- `nuxeo.streams.computation.processRecord`: Timer around processing of incoming record
+- `nuxeo.streams.computation.processTimer`: Timer around the computation timer processing
+- `nuxeo.streams.computation.running`: Number of active computation thread processing records or timers
```

Comment:
There, I didn't understand clearly that dimension means tags. I was tempted to replace `computation` keyword in the metric name by `audit/writer` for instance.

I don't know if we want to do something, maybe the part is understandable for SRE people and is enough.

Reply:
Ok, I will try to improve the dimensional metrics explanation (on top of the page), so it is clear that tags are used to express dimensions. The dimension here is `computation` that can be expressed by a tag `computation=audit/writer`.
The projected metric name depends on the monitoring system, you will have `nuxeo.stream.computation.audit-writer.running` for Graphite or `nuxeo.stream.computation.running{computation=audit-writer}` for Prometheus see the "Supported Monitoring Systems" section that gives example of projection.
If you browse metrics using a monitoring system it will more obvious.

Reply:
Thanks for the explanation, it is much clear now 👍 .